plugin/pkg/auth/authorizer/node/graph_populator.go

```diff
@@ -106,3 +123,31 @@ func (g *graphPopulator) deletePV(obj interface{}) {
 	}
 	g.graph.DeletePV(pv.Name)
 }
+
+func (g *graphPopulator) addVolumeAttachment(obj interface{}) {
+	g.updateVolumeAttachment(nil, obj)
+}
+
+func (g *graphPopulator) updateVolumeAttachment(oldObj, obj interface{}) {
+	attachment := obj.(*storagev1alpha1.VolumeAttachment)
+	if oldObj != nil {
+		// skip add if node name is identical
```

Comment:
This raises a question what happens if the node is not identical. New node is added, but the old one is not removed. Is it correct?

This question (and check) is only rhetorical, as `VolumeAttachment` validation blocks any changes in `VolumeAttachment.Spec` anyway

Reply:
graph.AddVolumeAttachment() first removes existing edges, then readds new edges